src/wallet/wallet.cpp

```diff
@@ -1515,6 +1550,12 @@ bool CWallet::DummySignTx(CMutableTransaction &txNew, const std::vector<CTxOut>
     for (const auto& txout : txouts)
     {
         CTxIn& txin = txNew.vin[nIn];
+        // If weight was provided, fill the input to that weight
+        if (coin_control && coin_control->HasInputWeight(txin.prevout)) {
```

Comment:
If the given value is "too low" as per https://github.com/bitcoin/bitcoin/pull/23201/commits/e7a318a8a88489477264060ccb332a719260fa67#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8R1515, it looks like logic uses just thinks it signed and returns `true`.

There's test coverage from the RPC perspective but ideally we'd catch this issue by returning false?

Reply:
Changed to return a bool.